### DISCUSSION TITLE
Counting sequence in RAM_DL

### DISCUSSION THREAD

**Discussion by cvu1** [RudolphRiedel/FT800-FT813#51] [Q&A] [answered]
Hi Rudolph,

I got this from brtcommunity about your post. (http://www.brtcommunity.com/index.php?topic=402.0)

> CLEAR_COLOR_RGB(0,0,0)
> CLEAR(1, 1, 1)
> LINE_WIDTH(80)
> BEGIN(RECTS)
> COLOR_RGB(255,215,0)
> TAG(100)
> VERTEX2F(1600, 1600)
> VERTEX2F(3200, 2240)
> TAG(110)
> VERTEX2F(1600, 2560)
> VERTEX2F(3200, 3200)
> COLOR_RGB(0,0,0)
> TAG(110)
> VERTEX2F(1632, 2592)
> VERTEX2F(3168, 3168)
> END()
> TAG(100)
> COLOR_RGB(0,0,0)
> CMD_TEXT(150, 120, 28, OPT_CENTER, "MUTE")
> TAG(110)
> COLOR_RGB(255,215,0)
> CMD_TEXT(150, 180, 28, OPT_CENTER, "NEXT")
> TAG(0)

How did you count this sequence uses 38 32bit words on the display-list. I only can count to 31  and I can't find any information how to count it.

Thanks,
vdc

**Comment by RudolphRiedel** [comment, marked as answer]
Hi,
all the simple commands are one 32 bit word, that makes 21 and leaves the two CMD_TXT.

Both CMD_TEXT have a text with a length of 4 chars.
EVE_cmd_text(int16_t x0, int16_t y0, int16_t font, uint16_t options, const char* text);
CMD_TEXT itself, coordinates, font and options are three 32 bit words.
And the text needs two 32 bit words since it has to end on 0x00 and be 32 bit aligned.

That makes five per CMD_TEXT and 31 in total, you are correct.

I am not sure what I was actually counting that day. :-)
And actually it even is only 29 since the first two commands are for the display list itself.

**Reply by RudolphRrr** [reply]
Hmm, this was not the 100% correct answer either since the question actually was about the amount of bytes in RAM_DL.
This sequence needs 31 32bit words to be transferred to the CMD_FIFO.
And this sequence needs 31 32bit words in RAM_DL, but the explanation for why it does was wrong.

CMD_TEXT gets translated by the command-coprocessor to 5 commands:
18	0x1f000001	BEGIN(BITMAPS)
19	0x8f46ce4d	VERTEX2II(122, 108, 28, 'M')
20	0x91a6ce55	VERTEX2II(141, 108, 28, 'U')
21	0x9346ce54	VERTEX2II(154, 108, 28, 'T')
22	0x9506ce45	VERTEX2II(168, 108, 28, 'E')

But only for coordinates below 511.

Above 511 this looks completely different:
18	0x22000000	SAVE_CONTEXT()
19	0x27000002	VERTEX_FORMAT(2)
20	0x0500001c	BITMAP_HANDLE(28)
21	0x1f000001	BEGIN(BITMAPS)
22	0x0600004d	CELL(77)
23	0x441401b0	VERTEX2F(2088, 432)
24	0x06000055	CELL(85)
25	0x443a01b0	VERTEX2F(2164, 432)
26	0x06000054	CELL(84)
27	0x445401b0	VERTEX2F(2216, 432)
28	0x06000045	CELL(69)
29	0x447001b0	VERTEX2F(2272, 432)
30	0x23000000	RESTORE_CONTEXT()

And "CMD_TEXT(150, 120, 28, OPT_CENTER, "MUTE!!!")" would still require to send 5 32bit words to the CMD_FIFO but would get translated to 8 32bit words.

Complications like this are why I usually do not count these myself and just print out the amount of bytes that are sent and the length of the resulting display-list.
